### PR TITLE
feat(transfer): adds create and revert a transfer

### PR DESCRIPTION
**Pull request by MbNakaya**
## Story
https://moippagamentos.atlassian.net/browse/DX-95
https://moippagamentos.atlassian.net/browse/DX-96

## Description
Adds create and revert a transfer features. Adds examples for all available transfers features.

## Tasks
- [x] add create a transfer feature
- [x] add tests to create a transfer (`TransferApiTest.java`)
- [x] add tests to create a transfer (`TransferRequestTest.java`)
- [x] document create a transfer example
- [x] document get transfer example
- [x] document list transfers example
- [x] add revert a transfer feature
- [x] add tests to revert a transfer (`TransferApiTest.java`)
- [ ] ~add tests to revert a transfer (`TransferRequestTest.java`)~ _that's not necessary_
- [x] document revert a transfer example

>**Observations**:
>* was necessary create another post method, at `Client.java`, to make the reverse transfer request;
>* the `MoipAccountRequest` class was removed from `ReceiverRequest.java` and added to `MoipAccountRequest.java` to be used in more files;
>* the `TransferInstrumentRequest.java` was created to be used at `TransferRequest.java`, in the creation of a transfer;
>* the `TransferRequest.java` was added to create a transfer;
>* was added a `toString()` method, at `Holder.java`, to fix the request response;
>* the `ReverseTransfer.java` resource was created to build the response and add the output methods;
>* the `TransferLinks.java` resource was created to catch the links that will be returned at the response;
>* some unnecessary imports was removed.

### CODE REVIEW

File: src/main/java/br/com/moip/api/TransferApi.java [hunk 1]
```diff
@@ -14,6 +15,10 @@ public TransferApi(final Client client) {
         this.client = client;
     }
 
+    public Transfer create(final TransferRequest transfer) { return client.post(TRANSFER_URL, transfer, Transfer.class); }
+
+    public Transfer reverse(final String id) { return client.post(TRANSFER_URL + "/" + id + "/" + "reverse", Transfer.class); }
```
Comment (bressani) [inline src/main/java/br/com/moip/api/TransferApi.java]: Essa concatenação "/" + "reverse" poderia não existir... Aqui também, caso queira, pode usar o String.format (https://docs.oracle.com/javase/6/docs/api/java/lang/String.html#format%28java.lang.String,%20java.lang.Object...%29)

File: src/test/java/br/com/moip/api/TransferApiTest.java [hunk 2]
```diff
@@ -1,8 +1,8 @@
 package br.com.moip.api;
 
 import br.com.moip.Client;
-import br.com.moip.resource.Transfer;
-import br.com.moip.resource.TransferStatus;
+import br.com.moip.request.TransferRequest;
+import br.com.moip.resource.*;
```
Comment (bressani) [inline src/test/java/br/com/moip/api/TransferApiTest.java]: É uma boa prática especificar os imports

File: src/main/java/br/com/moip/api/TransferApi.java (hunk 1 above)
Comment (MbNakaya) [reply src/main/java/br/com/moip/api/TransferApi.java]: Done

File: src/test/java/br/com/moip/api/TransferApiTest.java (hunk 2 above)
Comment (MbNakaya) [reply src/test/java/br/com/moip/api/TransferApiTest.java]: Done

File: src/main/java/br/com/moip/request/TransferInstrumentRequest.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        return this;
+    }
+
+    public enum Method{
+        BANK_ACCOUNT, MOIP_ACCOUNT;
+    }
+
+    @Override
+    public String toString(){
+        if(method == Method.BANK_ACCOUNT)
```
Comment (joaolucasl) [inline src/main/java/br/com/moip/request/TransferInstrumentRequest.java]: Dá pra simplificar isso aqui colocando o string builder numa variável, buildando até a chegar na parte do `bankAccount/moipAccount` onde vc faz um if e depois continuar. Assim tá bem legível, mas n tem pq repetir, acho.

File: src/test/java/br/com/moip/api/TransferApiTest.java [hunk 4]
```diff
@@ -20,11 +23,35 @@
...
     @Before
     public void setUp() {
         Client client = clientFactory.client(player.getURL("").toString());
         api = new TransferApi(client);
+        transferRequest = new TransferRequest();
+    }
+
+    @Play("transfers/create")
+    @Test
+    public void shouldCreateATransfer() {
```
Comment (joaolucasl) [inline src/test/java/br/com/moip/api/TransferApiTest.java]: Tenta criar uns casos que falhe a criação tmbm e checa isso pros testes serem mais abrangentes :)

Comment (MbNakaya) [reply src/test/java/br/com/moip/api/TransferApiTest.java]: Done

File: src/main/java/br/com/moip/request/TransferInstrumentRequest.java (hunk 3 above)
Comment (MbNakaya) [reply src/main/java/br/com/moip/request/TransferInstrumentRequest.java]: Done

File: src/main/java/br/com/moip/request/TransferInstrumentRequest.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        return this;
+    }
+
+    public enum Method{
+        BANK_ACCOUNT, MOIP_ACCOUNT;
+    }
+
+    @Override
+    public String toString(){
+        StringBuilder stringBuilder = new StringBuilder("TransferInstrumentRequest{")
```
[the remark this answers is not in the document]
Comment (MbNakaya) [reply src/main/java/br/com/moip/request/TransferInstrumentRequest.java]: Done

File: src/main/java/br/com/moip/resource/Transfer.java [hunk 6]
```diff
@@ -82,8 +85,22 @@ public void setRole(Role role) {
         PAYER
     }
 
+    public TransferLinks getLinks() { return _links; }
+    public void setLinks(TransferLinks _links) { this._links = _links; }
+
     @Override
     public String toString() {
+        if(status == TransferStatus.REVERSED)
```
[the remark this answers is not in the document]
Comment (MbNakaya) [reply src/main/java/br/com/moip/resource/Transfer.java]: Done

**Comment by MbNakaya** [comment]
resolves #76

**Pull request merged**